agent/docker.go

```diff
@@ -0,0 +1,152 @@
+package agent
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/armon/circbuf"
+)
+
+// DockerClient is a simplified client for the Docker Engine API
+// to execute the health checks and avoid significant dependencies.
+// It also consumes all data returned from the Docker API through
+// a ring buffer with a fixed limit to avoid excessive resource
+// consumption.
+type DockerClient struct {
+	network string
+	addr    string
+	baseurl string
+	maxbuf  int64
+	client  *http.Client
+}
+
+func NewDockerClient(host string, maxbuf int64) (*DockerClient, error) {
+	if host == "" {
+		host = DefaultDockerHost
+	}
+	p := strings.SplitN(host, "://", 2)
+	if len(p) == 1 {
```

Comment:
I don't think that `len(p)` can ever be zero unless you specify `0` as the last argument. If the split string isn't found you get `host` back.